dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3PutJob.java

```diff
@@ -115,12 +116,12 @@ public void executeJob() throws Exception {
                     final String ds3FileName = PathUtil.toDs3Path(targetDir, ds3ObjPath);
                     fileMap.put(ds3FileName, path1);
                 });
-                directories.stream().forEach(path -> {
+                directories.forEach(path -> {
                     try {
                         partOfDirBuilder.add(path);
                         expandedPaths.putAll(path, Files.walk(path).filter(child -> !Files.isDirectory(child)).collect(GuavaCollectors.immutableList()));
-                        final String ds3ObjPath = getDs3ObjectPath(path, path, true, files.size(), directories.size());
-                        final String ds3FileName = PathUtil.toDs3Path(targetDir, ds3ObjPath);
+                        final String ds3ObjPath = getDs3ObjectPath(path, path, true, files.size(), directories.size()); // TODO what is this code doing exactly
```

Comment:
Not knowing what getDs3ObjectPath does, it seems a little odd that path is passed as 2 different parameters.